packages/api/src/external/carequality/patient.ts

```diff
@@ -0,0 +1,70 @@
+import { IHEGateway, PatientDiscoveryResponse } from "@metriport/ihe-gateway-sdk";
+import { Organization } from "../../domain/medical/organization";
+import { Patient } from "../../domain/medical/patient";
+import { capture } from "../../shared/notifications";
+import { makeIheGatewayAPI } from "./api";
+import { toFHIR } from "../fhir/patient";
+import { patientToIheGateway } from "./patient-conversion";
+import { setCarequalityId } from "./patient-external-data";
+
+const createContext = "cq.patient.create";
+
+export async function discover(
+  patient: Patient,
+  organization: Organization,
+  facilityId: string
+): Promise<void> {
+  let iheGateway: IHEGateway | undefined;
+  try {
+    console.log(`IHE Gateway patient discovery - M patientId ${patient.id}`);
+
+    iheGateway = makeIheGatewayAPI();
+
+    const fhirPatient = toFHIR(patient);
+
+    const iheGatewayRequest = patientToIheGateway({
+      patient: fhirPatient,
+      cxId: patient.cxId,
+      xcpdGateways: [], // TODO ADD XCPD GATEWAYS WHEN WE HAVE THEM
+      facilityNPI: facilityId,
+      orgName: organization.data.name,
+      orgOid: organization.oid,
+    });
+
+    iheGateway.startPatientDiscovery(iheGatewayRequest);
+  } catch (err) {
+    console.error(
+      `Failure while starting patient discovery for ${patient.id} @ IHE Gateway: `,
+      err
+    );
+    capture.error(err, {
+      extra: {
+        facilityId,
+        patientId: patient.id,
+        context: createContext,
+      },
+    });
+    throw err;
+  }
+}
+
+export async function handleDiscoverResponse(
+  patientDiscoveryResponse: PatientDiscoveryResponse
+): Promise<void> {
+  const { cxId, patientId, xcpdPatientId, patientMatch, gateway } = patientDiscoveryResponse;
+
+  if (patientMatch) {
+    await setCarequalityId({
+      patientId,
+      cxId,
+      carequalityPatientId: xcpdPatientId?.id ?? "",
+      carequalityPatientSystemId: xcpdPatientId?.system ?? "",
+    });
+  } else {
+    // WE SHOULD EXPECT THIS TO HAPPEN SO SHOULD WE ADD SENTRY MESSAGE?
+    // DONT WANT TO BOMBARD ALERTS CHANNEL
```

Comment:
Unless there's something for us to do, I think a console is fine (might even be too much, we also don't want to make logs unusable.

Reply:
Yea i was thinking at least a log for debugging purposes. Lets say a patient has bad coverage we can look at the logs and see where they werent able to get a match

Reply:
I like that, but how many entries are we expecting to get on the logs? If its 10, that's fine, if its +1,000 that's likely too much (not saying we shouldn't keep track of it, just saying logs wouldn't be a good solution in this case - maybe a DynamoDB table?).